src/lib/core/sample_sort.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include <stddef.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+#define SAMPLE_SORT_THREADS_MAX 256
+
+typedef int
+(*compare_arg_f)(const void *a, const void *b, void *arg);
```

Comment:
The type name is too generic. Better define this type in the function arguments.

Reply:
Changed name to more specific:
```c
typedef int
(*tt_sort_compare_f)(const void *a, const void *b, void *arg);
```